.github/workflows/doc.yml

```diff
@@ -0,0 +1,46 @@
+name: "Run: Build documentation"
+
+permissions: {}
+
+concurrency:
+  group: "${{ github.workflow }}-${{ github.ref }}-doc-${{ github.run_id }}"
+  cancel-in-progress: true
+
+on:
+  pull_request:
+    types: [ opened, synchronize, reopened ]
+    paths:
+      - 'doc/**'
+  push:
+    paths:
+      - 'doc/**'
+
+jobs:
+  doc:
+    permissions:
+      contents: write
+    concurrency:
+      group: "doc-${{ inputs.run_id }}"
+      cancel-in-progress: true
+    uses: antirs/sh.wrap/.github/workflows/actions.yml@actions
```

Comment:
Building docker images every time is time consumint. Push them to docker hub and reuse.